istioctl/pkg/install/pre-check.go

```diff
@@ -0,0 +1,159 @@
+package install
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	authorizationapi "k8s.io/api/authorization/v1beta1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
+)
+
+const (
+	minK8SVersion = "1.11"
+)
+
+type targetResource struct {
+	namespace string
+	group     string
+	version   string
+	name      string
+}
+
+func installPreCheck(istioNamespaceFlag *string, restClientGetter resource.RESTClientGetter, writer io.Writer) error {
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "Checking the cluster to make sure it is ready for Istio installation...\n")
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "kubernetes-api\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+	kubeClient, err := createKubeClient(restClientGetter)
+	if err != nil {
+		return fmt.Errorf("failed to initialize the k8s client: %v", err)
+	}
+	fmt.Fprintf(writer, "can initialize the k8s client\n")
+	v, err := kubeClient.Discovery().ServerVersion()
+	if err != nil {
+		return fmt.Errorf("failed to query the Kubernetes API Server: %v", err)
+	}
+	fmt.Fprintf(writer, "can query the Kubernetes API Server\n")
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "kubernetes-version\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+
+	if !checkKubernetesVersion(v) {
+		return fmt.Errorf("The Kubernetes API version: %v is lower than the minimum version: "+minK8SVersion, v)
+	}
+	fmt.Fprintf(writer, "is running the minimum Kubernetes API version\n")
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "Istio-existence\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+	ns, _ := kubeClient.CoreV1().Namespaces().Get(*istioNamespaceFlag, meta_v1.GetOptions{})
+	if ns != nil {
+		return fmt.Errorf("control plane namespace: %v already exist", *istioNamespaceFlag)
```

Comment:
"Istio cannot be installed because the Istio namespace %q is already in use."

Reply:
The grammar of "control plane namespace: %v already exist" is still incorrect.  It should be `already exists`.  The message I suggest, `"Istio cannot be installed because the Istio namespace %q is already in use."` might be easier to understand.